cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2053,6 +2054,13 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
             for (GenericKafkaListener listener : loadBalancerListeners) {
                 String bootstrapServiceName = ListenersUtils.backwardsCompatibleBootstrapServiceName(name, listener);
 
+                // When the listener is load balancer and useCreateBootStrapService is false,
+                // we will skip the creation of the external load balancer to save the cost.
+                if (KafkaListenerType.LOADBALANCER == listener.getType() && listener.getConfiguration() != null
+                        && !listener.getConfiguration().getCreateBootstrapService()) {
+                    continue;
+                }
```

Comment:
I'm afraid this would be too simple :-/. It skips the creation of the ListenerStatus which we definitely do not want to skip. We would want it at least without the address, but ideally with the address constructed from the broker services / loadbalancers.

Reply:
Sorry, I don't understand what you mentioned above.  You mean we want to keep the headless service for the bootstrap service, but don't assign a load balancer IP for it? Here is my test example:

If we skip the bootstrap service, I got the following services. The external bootstrap service is skipped.

<img width="822" alt="image" src="https://user-images.githubusercontent.com/95244574/149237938-d62fdc08-a004-430e-a9fd-aeb7b95d8167.png">

Do we still want to create the external-bootstrap service, but without an External Load Balancer assigned? I guess my confusion is where is the ListenerStatus defined.

Reply:
put it in another way, what should the expect output be.

Reply:
When you look at the deployed Kafka cluster, in the status section you should see something like this:

```yaml
    - addresses:
      - host: 192.168.1.72
        port: 9094
      bootstrapServers: 192.168.1.72:9094
      certificates:
      - |
        -----BEGIN CERTIFICATE-----
        ...
        -----END CERTIFICATE-----
      type: external
```

That is all generated here on the lines 2077-2084. Your code basiclly just skips this, so there would be no record about this listener in the status. I think we should think about how to handle that. Ideally, we would want there multiple addresses instead of the _per-broker_ load balancers. Not sure how easy that would be. But if nothing else, we would probably want to at least have the information abott the listener without the address.

Reply:
I got your point now, thanks for the explanation! Let me look at the code again.

Reply:
So we would like to change the listener status to this (if we skip the creation of the bootstrap service)? use the above example:

``` 
     - addresses:
      bootstrapServers: 192.168.1.72:9094 
      certificates:
      - |
        -----BEGIN CERTIFICATE-----
        ...
        -----END CERTIFICATE-----
      type: external
```

It seems like a better idea as it aligns with Apache Kafka's bootstrap services, so it's widely recognized by the kafka users.

Am I correct?

Reply:
Yes. I think I would be flexible if it would be too complicated. But ideally we would probably want something like this when the bootstrap service is disabled:

```yaml
    - addresses:
        - host: 192.168.1.72
          port: 9094
        - host: 192.168.1.73
          port: 9094
        - host: 192.168.1.74
          port: 9094
      bootstrapServers: 192.168.1.72:9094,192.168.1.73:9094,192.168.1.74:9094 
      certificates:
        - |
          -----BEGIN CERTIFICATE-----
          ...
          -----END CERTIFICATE-----
      type: external
```

Where `192.168.1.72:9094,192.168.1.73:9094,192.168.1.74:9094 ` are the addresses of the loadbalancers corresponding to the individual brokers.

Reply:
Got it, thanks! It shouldn't be hard. let me try again and update the PR later.